mx.py

```diff
@@ -7089,7 +7126,14 @@ def _returncode(status):
             return _returncode(status)
         remaining = end - time.time()
         if remaining <= 0:
-            abort('Process timed out after {0} seconds: {1}'.format(timeout, ' '.join(args)))
+            msg = 'Process timed out after {0} seconds: {1}'.format(timeout, ' '.join(args))
+            if nonZeroIsFatal:
+                abort(msg)
+            else:
+                log(msg)
+                for p, _ in _currentSubprocesses:
```

Comment:
`abort` would kill all sub-processes. I tried to keep the semantics as similar as possible, so that the only difference is that the current process survives. However, I see your point and I am going to change this in my next push.